src/m365/teams/commands/channel/channel-add.ts

```diff
@@ -80,16 +80,16 @@ class TeamsChannelAddCommand extends GraphCommand {
           return `${args.options.teamId} is not a valid GUID`;
         }
 
-        if (args.options.type && ['standard', 'private'].indexOf(args.options.type) === -1) {
+        if (args.options.type && ['standard', 'private', 'shared'].indexOf(args.options.type) === -1) {
           return `${args.options.type} is not a valid type value. Allowed values standard|private.`;
         }
 
-        if (args.options.type === 'private' && !args.options.owner) {
-          return 'Specify owner when creating a private channel.';
+        if ((args.options.type === 'private' || args.options.type === 'shared') && !args.options.owner) {
+          return `Specify owner when creating a ${args.options.type} channel.`;
         }
 
-        if (args.options.type !== 'private' && args.options.owner) {
-          return 'Specify owner only when creating a private channel.';
+        if ((args.options.type !== 'private' && args.options.type !== 'shared') && args.options.owner) {
+          return `Specify owner only when creating a ${args.options.type} channel.`;
```

Comment:
We shouldn't write the selected type here.
Otherwise we'll get `Specify owner only when creating a undefined channel.` when creating a standard channel with owner.

```suggestion
          return `Specify owner only when creating a private or shared channel.`;
```